pkg/minishift/openshift/oc.go

```diff
@@ -68,3 +90,114 @@ func AddContextForProfile(profile string, ip string, username string, namespace
 	}
 	return nil
 }
+
+// Get the route for service
+func GetServiceURL(service, namespace string, https bool) (string, error) {
+	urlScheme := "http://"
+	if https {
+		urlScheme = "https://"
+	}
+
+	if namespace == "default" {
+		return "", errors.New("Namespace need to be specified with -n option")
+	}
+
+	if !isProjectExists(namespace) {
+		return "", errors.New(fmt.Sprintf("Namespace %s doesn't exits", namespace))
+	}
+
+	cmdName := instanceState.Config.OcPath
+	cmdArgText := fmt.Sprintf("get route/%s -n %s --config=%s %s", service, namespace, systemKubeConfigPath, URLCustomCol)
```

Comment:
hmmm, the idea with the cached cube config was that it becomes part of some utility function. So that all our internal _oc_ usage (unless specified explicitly) does use the _--config_ flag. Now you just apply it for your use case, so other code does not benefit and still needs to be adjusted. Also new code will need to remember to apply it. 

_Runner_ is just a plain wrapper around _exec.Command_. What we need is a wrapper around _oc_, e.g. OcRunner. This runner handles get configured with the path to oc and then offers for example to methods to execute a oc command. One which applies the --config and one without. All _oc_ usage should go through there then. This ensures, the right parameters are applied and we can for example instrument the _oc_ calls and take timing in order to find out how long oc calls take.

This was the initial idea of _oc.go_. A wrapper around _oc_ which then internally uses a _Runner_.

Let me know what you think and whether you are going to address this. If not I will take this as part of issue #177. I need a wrapper around _oc_ now as well to continue.